### ISSUE TITLE
TypeError: Cannot read property 'defaultEditorFile' of undefined amplify when updating function permission and editing it locally.

### ISSUE THREAD

**Issue by kylekirkby** [aws-amplify/amplify-cli#3821]
**Describe the bug**
After running `amplify update function` and changing the permissions this function requires, the following error appears after trying to edit the function locally:
```
TypeError: Cannot read property 'defaultEditorFile' of undefined
    at openEditor (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/lib/provider-utils/awscloudformation/index.js:97:38)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Object.updateResource (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/lib/provider-utils/awscloudformation/index.js:84:9)
    at async Object.executeAmplifyCommand (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/lib/index.js:123:5)
    at async executePluginModuleCommand (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:131:13)
    at async Object.executeCommand (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/lib/execution-manager.js:16:9)
    at async Object.run (/home/acidburn/.nvm/versions/node/v12.11.1/lib/node_modules/@aws-amplify/cli/lib/index.js:43:9)
```

**Amplify CLI Version**
```
4.17.2
```

**To Reproduce**
Add a function. Update the function and change some of the permissions the function needs to execute. Enter `Y` to edit the function locally and then see the following error:
```
TypeError: Cannot read property 'defaultEditorFile' of undefined
```

**Expected behavior**
Open up the default editor to edit the function as with when creating the function (this works strangely).

**Desktop (please complete the following information):**
 - Ubuntu 18.04
 - Node v 12.11.1

**Comment by stevelizcano** [comment]
+1 here,

You can reproduce by adding auth to any project, and specifying a custom welcome function. Accept defaults, and signup will fail with an error. Editing the the function will also fail.

**Comment by attilah** [comment]
The fix has been release with Amplify CLI 4.18.0
